tests/unit/dataactvalidator/test_b18_object_class_program_activity_1.py

```diff
@@ -0,0 +1,45 @@
+from factory import fuzzy
+from tests.unit.dataactcore.factories.staging import ObjectClassProgramActivityFactory
+from tests.unit.dataactvalidator.utils import number_of_errors, query_columns
+
+
+_FILE = 'b18_object_class_program_activity_1'
+
+not_req_dict = {
+        "submission_id":fuzzy.FuzzyInteger(9999),
+        "object_class":fuzzy.FuzzyInteger(1000,9999),
+        "by_direct_reimbursable_fun":""
+}
+
+req_present_dict = {
+        "submission_id":fuzzy.FuzzyInteger(9999),
+        "object_class":fuzzy.FuzzyInteger(999),
+        "by_direct_reimbursable_fun":"d"
+}
+
+req_absent_dict = {
+        "submission_id":fuzzy.FuzzyInteger(9999),
+        "object_class":fuzzy.FuzzyInteger(999),
+        "by_direct_reimbursable_fun":""
+}
+
+def test_column_headers(database):
+    expected_subset = {"row_number", "object_class", "by_direct_reimbursable_fun"}
+    actual = set(query_columns(_FILE, database))
+    assert expected_subset <= actual
+
+def test_success(database):
+    """ Test that a four digit object class with no flag is a success, and a three digit object class with a flag is a success"""
+    not_req_ocpa = ObjectClassProgramActivityFactory(**not_req_dict)
+    req_present_ocpa = ObjectClassProgramActivityFactory(**req_present_dict)
```

Comment:
This may just be a style thing, but one of the goals of factory methods is that folks can read the tests with minimal distractions/lookups. Why not remove the module-level variables and spell everything out here? I think you can do:
```python
not_req_ocpa = ObjectClassProgramActivityFactory(
    by_direct_reimbursable_fun='', object_class=str(randint(1000, 10000)))
req_present_ocpa = ObjectClassProgramActivityFactory(
    by_direct_reimbursable_fun='d', object_class=str(randint(100, 1000)))
```